### PR TITLE
Document generated methods

### PR DESCRIPTION
**Pull request by danburkert**
Extracted just the method getter/setter changes from #234, plus a few improvements.

### CODE REVIEW

File: prost-derive/src/field/map.rs [hunk 1]
```diff
@@ -256,10 +256,18 @@ impl Field {
                 quote!()
             };
 
+            let get_doc = format!(
+                "Returns the enum value for the corresponding key in `{}`, \
+                 or `None` if the entry does not exist or it is not a valid enum value.",
```
Comment (danburkert) [inline prost-derive/src/field/map.rs]:
```suggestion
                 or `None` if the entry does not exist or is not a valid enum value.",
```

**Comment by danburkert** [comment]
@dunmatt apologies, I merged this via 'squash', which I did not realize would overwrite the git author metadata.  I've corrected that mistake on master by forcibly rewriting the commits.

**Comment by dunmatt** [comment]
Lol, I'm not worried about credit (but on behalf of those who do, thanks
for offering to fix it!), I care about things working well!

**Comment by danburkert** [comment]
No worries, I personally feel strongly that the PR credit should be accurately reflected in the author metadata.  There are very few things which I'll rewrite history for, but this is one of them!

**Pull request merged**